### PR TITLE
deprecate addon field from the operator

### PR DESCRIPTION
**Pull request by richardwxn**
in 1.7 we deprecated the telemetry addons and in 1.8 we deprecate istiocoredns as well so we can mark this whole field as deprecated.

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]
what is the user result of this? just a doc change?

**Comment by richardwxn** [comment]
@howardjohn it is doc change, and for telemetry addon specifically, we would fail hard during runtime of installation

**Comment by richardwxn** [comment]
@linsun @smawson Can you help review?

**Comment by linsun** [comment]
@richardwxn can you make sure we have release notes indicate the deprecation?

Otherwise LGTM.

File: operator/v1alpha1/operator.proto [hunk 1]
```diff
@@ -99,7 +99,7 @@ message IstioOperatorSpec {
     IstioComponentSetSpec components = 50;
 
     // Extra addon components which are not explicitly specified above.
-    map<string, ExternalComponentSpec> addon_components = 51;
+    map<string, ExternalComponentSpec> addon_components = 51 [deprecated = true];
```
Comment (smawson) [inline operator/v1alpha1/operator.proto:102]: Can you add a comment here indicating that Istio will no longer manage external components and that users should move to managing these themselves?

Comment (richardwxn) [reply operator/v1alpha1/operator.proto:104]: added

**Review by smawson — APPROVED** [review]

**Pull request merged**